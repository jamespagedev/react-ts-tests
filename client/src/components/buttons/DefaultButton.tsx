import React from 'react';

interface Props {
  btnText: string;
  updateParentState: () => void;
}

function DefaultButton({btnText, updateParentState}: Props): JSX.Element {
  function handleButtonClick() {
    console.log(btnText);
    updateParentState();
  }
  return <button onClick={handleButtonClick}>{btnText}</button>;
}

export default DefaultButton;
