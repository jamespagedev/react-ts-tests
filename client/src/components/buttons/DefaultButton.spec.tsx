import React from 'react';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import DefaultButton from './DefaultButton';

// ===========================================================================
// =============================== setup tests ===============================
// ===========================================================================
// -------------------------------- variables --------------------------------
const btnText = 'Click Me';

// ------------------------------ mock functions -----------------------------
const mockUpdateParentStateFn = jest.fn();
const logSpy = jest.spyOn(console, 'log');
// ---------------------------------- types ----------------------------------
interface TestContainerProps {
  btnText: string;
}

// -------------------------------- containers -------------------------------
function TestContainer({btnText}: TestContainerProps): JSX.Element {
  return <DefaultButton btnText={btnText} updateParentState={mockUpdateParentStateFn} />;
}

// ===========================================================================
// =============================== start tests ===============================
// ===========================================================================
describe('DefaultButton', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('matches snapshot', () => {
    const {container} = render(<TestContainer btnText={btnText} />);
    expect(container).toMatchSnapshot();
  });
  it('click invokes mockHandleButtonClickFn function', () => {
    render(<TestContainer btnText={btnText} />);
    const btn = screen.getByText(btnText);
    userEvent.click(btn);
    expect(mockUpdateParentStateFn).toHaveBeenCalledTimes(1);
  });
  it('click ensures console.log runs', () => {
    render(<TestContainer btnText={btnText} />);
    const btn = screen.getByText(btnText);
    userEvent.click(btn);
    expect(logSpy).toHaveBeenCalledTimes(1);
  });
});
