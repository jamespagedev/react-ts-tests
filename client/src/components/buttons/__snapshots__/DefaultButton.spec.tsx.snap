// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultButton matches snapshot 1`] = `
<div>
  <button>
    Click Me
  </button>
</div>
`;
